src/App.tsx

```diff
@@ -74,35 +78,53 @@ function App() {
   const [darkMode, setDarkMode] = useState<boolean>(false)
   const fileInputRef = useRef<HTMLInputElement>(null)
 
-  const moveItem = useCallback((group: 'today' | 'tomorrow' | 'upcoming', dragIndex: number, hoverIndex: number) => {
+  const moveItem = useCallback((sourceGroup: 'today' | 'tomorrow' | 'upcoming', targetGroup: 'today' | 'tomorrow' | 'upcoming', dragIndex: number, hoverIndex: number) => {
     setTodos((prevTodos) => {
       const newTodos = [...prevTodos];
-      let groupTodos: Todo[];
-      if (group === 'today') {
-        groupTodos = newTodos.filter(isDueOrOverdue);
-      } else if (group === 'tomorrow') {
-        groupTodos = newTodos.filter(todo => isTomorrow(todo.dueDate));
+      let sourceTodos: Todo[];
+      let targetTodos: Todo[];
+
+      if (sourceGroup === 'today') {
+        sourceTodos = newTodos.filter(isDueOrOverdue);
+      } else if (sourceGroup === 'tomorrow') {
+        sourceTodos = newTodos.filter(todo => isTomorrow(todo.dueDate));
       } else {
-        groupTodos = newTodos.filter(todo => !isDueOrOverdue(todo) && !isTomorrow(todo.dueDate));
```

Comment:
I get this error when I try to drag from one group to another:
```
App.tsx:112 Uncaught TypeError: Cannot set properties of undefined (setting 'dueDate')
    at App.tsx:112:21
    at basicStateReducer (react-dom_client.js?v=2ef889ff:11703:49)
    at updateReducer (react-dom_client.js?v=2ef889ff:11794:30)
    at updateState (react-dom_client.js?v=2ef889ff:12021:18)
    at Object.useState (react-dom_client.js?v=2ef889ff:12753:24)
    at useState (chunk-DRWLMN53.js?v=2ef889ff:1066:29)
    at App (App.tsx:75:29)
    at renderWithHooks (react-dom_client.js?v=2ef889ff:11548:26)
    at updateFunctionComponent (react-dom_client.js?v=2ef889ff:14582:28)
    at beginWork (react-dom_client.js?v=2ef889ff:15924:22)
(anonymous) @ App.tsx:112
basicStateReducer @ react-dom_client.js?v=2ef889ff:11703
updateReducer @ react-dom_client.js?v=2ef889ff:11794
updateState @ react-dom_client.js?v=2ef889ff:12021
useState @ react-dom_client.js?v=2ef889ff:12753
useState @ chunk-DRWLMN53.js?v=2ef889ff:1066
App @ App.tsx:75
renderWithHooks @ react-dom_client.js?v=2ef889ff:11548
updateFunctionComponent @ react-dom_client.js?v=2ef889ff:14582
beginWork @ react-dom_client.js?v=2ef889ff:15924
callCallback2 @ react-dom_client.js?v=2ef889ff:3674
invokeGuardedCallbackDev @ react-dom_client.js?v=2ef889ff:3699
invokeGuardedCallback @ react-dom_client.js?v=2ef889ff:3733
beginWork$1 @ react-dom_client.js?v=2ef889ff:19765
performUnitOfWork @ react-dom_client.js?v=2ef889ff:19198
workLoopSync @ react-dom_client.js?v=2ef889ff:19137
renderRootSync @ react-dom_client.js?v=2ef889ff:19116
performConcurrentWorkOnRoot @ react-dom_client.js?v=2ef889ff:18678
workLoop @ react-dom_client.js?v=2ef889ff:197
flushWork @ react-dom_client.js?v=2ef889ff:176
performWorkUntilDeadline @ react-dom_client.js?v=2ef889ff:384
Show 19 more frames
Show less
App.tsx:112 Uncaught TypeError: Cannot set properties of undefined (setting 'dueDate')
    at App.tsx:112:21
    at basicStateReducer (react-dom_client.js?v=2ef889ff:11703:49)
    at updateReducer (react-dom_client.js?v=2ef889ff:11794:30)
    at updateState (
```
    
Maybe we can add some debugging statements to troubleshoot this?

Reply:
Now I'm getting this error:
Uncaught ReferenceError: Cannot access 'isDueOrOverdue' before initialization

Reply:
Can we please add .aider* to our .gitignore file?